packages/cli/internal/pkg/cli/workflow/manager.go

```diff
@@ -348,6 +357,67 @@ func (m *Manager) parseInputToArguments() {
 	m.arguments = []string{arguments}
 }
 
+func (m *Manager) initializeTempDir() {
+	if m.err != nil {
+		return
+	}
+	var err error
+	m.tempPath, err = createTempDir("", "workflow_*")
+	log.Debug().Msgf("created temp directory at: '%s'", m.tempPath)
+	if err != nil {
+		m.err = err
+		return
+	}
+	log.Debug().Msgf("recursively copying content of '%s' to '%s'", m.path, m.tempPath)
+	err = copyFileRecursivelyToLocation(m.tempPath, m.path)
+	if err != nil {
+		log.Error().Err(err)
+		m.err = err
+		return
+	}
+}
+
+func (m *Manager) deleteTempDir() {
+	if m.tempPath == "" {
+		return
+	}
+	err := removeAll(m.tempPath)
+	if err != nil {
+		log.Warn().Msgf("Failed to delete temporary folder '%s'", m.tempPath)
+	}
+}
+
+// writeTempManifest writes the inputsFile included in the command line to the temporary MANIFEST.json located in temp directory
+// This function is only called if there is a path included in the command line with the --inputsFile flag
+func (m *Manager) writeTempManifest() {
+	if m.err != nil || m.inputsPath == "" {
+		return
+	}
+	m.manifestPath = filepath.Join(m.tempPath, manifestFilename)
+	log.Debug().Msgf("Reading temp manifest %s", m.manifestPath)
+	bytes, err := m.Storage.ReadAsBytes(m.manifestPath)
+	if err != nil {
+		m.err = err
+		return
+	}
+	var data ManifestProps
+	if err := json.Unmarshal(bytes, &data); err != nil {
+		m.err = err
+		return
+	}
+	data.InputFileURLs = append(data.InputFileURLs, filepath.Base(m.inputsPath))
+	bytes, err = json.Marshal(data)
+	if err != nil {
+		m.err = err
+		return
+	}
+	err = m.Storage.WriteFromBytes(m.manifestPath, bytes)
+	if err != nil {
+		m.err = err
+		return
+	}
```

Comment:
considering that we are writing inputFiles into MANIFEST.JSON and upload to s3 bucket. can we also double check whether we specify the correct path? After we download MANIFEST.JSON file from s3 from WesAdapter, we can correctly refer it?